extensions/env-injector/src/main.rs

```diff
@@ -43,20 +43,51 @@ async fn snapdapi_req() -> Result<serde_json::Value, Box<dyn Error + Send + Sync
     Ok(serde_json::from_slice(&body)?)
 }
 
+fn process_env(env: &serde_json::Value) -> HashMap<String, String> {
+    let obj = env.as_object()
+        .ok_or("Expected an object (JSON input)").unwrap();
+    let mut map = HashMap::new();
+
+    for (k, v) in obj {
+        if v.is_object() || v.is_array() {
+            eprintln!(
+                "ERROR: Invalid value detected.\n\
+                    Key: {}\n\
+                    Reason: Environment variable names must not contain dots.\n\
+                    Action: Skipped\n",
+                v
+            );
+            continue;
+        }
+
+        let key = k.to_uppercase().replace("-", "_");
+        let value = match v {
+            serde_json::Value::String(s) => s.clone(),
+            serde_json::Value::Number(n) => n.to_string(),
+            serde_json::Value::Bool(b) => b.to_string(),
+            _ => {
+                eprintln!(
+                    "ERROR: Invalid type for environment variable value detected.\n\
+                        val: {}\n\
+                        Reason: It must be a string, number or boolean\n\
+                        Action: Skipped\n",
+                    k
+                );
+                continue;
+            },
```

Comment:
Actually, there is a tiny difference, the fist just checks if there isn't dots (which is a nested JSON) or arrays passed.

The second one does the type conversion. But I think it would be better to just always convert to a string.

Reply:
If it is neither a nested object nor an array, then it is a primitive type. No?